pmtiles/bucket.go

```diff
@@ -242,7 +244,59 @@ func (ba BucketAdapter) NewRangeReaderEtag(ctx context.Context, key string, offs
 	return reader, resultETag, status, nil
 }
 
-func (ba BucketAdapter) Close() error {
+func (ba S3BucketAdapter) Close() error {
+	return ba.Bucket.Close()
+}
+
+type AzureBucketAdapter struct {
```

Comment:
Is there a way to centralize this logic in the existing Bucket Adapter instead of creating separate implementations for each of (Azure, S3, etc)? That will be more maintainable in the long term and takes better advantage of gocloud.

Reply:
I've tried that in this updated version, there is nothing left in a "`BaseBucketAdapter`" cause everything we do here is using the prtovider specifc APIs (`As()`), so it's not generic anymore.

The only common left is now in `handleReaderResponse()`

I think I almost prefer the explicit version with both adapters and a few repetitions,